File: PYTHON/06searching/binarySearch.py
def binarySearch(a, length):
    key = int(input("Enter the value you want to search: "))
    f = False
    low = 0
    high = length - 1

    while low <= high:
        mid = (low + high) // 2
        if a[mid] == key:
            print("key found at index", mid)
            f = True
            break
        elif a[mid] < key:
            low = mid + 1
        else:
            high = mid - 1

    if not f:
        print(key, 'not found in', a)
        a.append(key)
        a.sort()  # Ensure the list is sorted after appending
        length = len(a)  # Update the length of the list
        print('new list is:', a)
        binarySearch(a, length)

a = [33, 77, 44, 11, 88, 44, 33, 22, 88]
a.sort()  # Ensure the list is sorted before starting binary search
length = len(a)
print('list is:', a)
binarySearch(a, length)

File: C/searching/binary_search.c
//binary search

#include<stdio.h>
int main(){
	int a[] ={11,44,33,77,45};
	int len = sizeof(a)/sizeof(a[0]);
	int i,j,key,low,mid,high,t,f=0;
	printf("before sort\n");
	for(i=0;i<len;i++){
		printf("%d ",a[i]);
	}
	printf("\nafter sort\n");
	for(i=0;i<len;i++){
		for(j = i+1 ; j<len;j++){
			if(a[i]>a[j]){
				t= a[i];
				a[i] = a[j];
				a[j]= t;
			}
		}
	}
	for(i=0;i<len;i++){
		printf("%d ",a[i]);
	}
	printf("\n enter value to be searched\n");
	scanf("%d",&key);
	
	low = 0;
	high = len-1;
	mid = (low+high)/2;
	
	while(low<=high){
		if(a[mid]==key){
			mid++;
			printf("\nposition = %d\n",mid);
			f= 1;
			break;
		}
		else if(a[mid]>key){
			high = mid - 1;
		}
		else if(a[mid]<key){
			high = mid + 1;
		}
		mid = (low+high)/2;
	}
	if(f == 0){
		printf("\nnot found");
	}
	return 0;
	
}
